src/main/thrift/parquet.thrift

```diff
@@ -237,6 +237,78 @@ struct SizeStatistics {
    3: optional list<i64> definition_level_histogram;
 }
 
+/**
+ * Interpretation for edges of GEOMETRY logical type, i.e. whether the edge
+ * between points represent a straight cartesian line or the shortest line on
+ * the sphere.
+ */
+enum Edges {
+  PLANAR = 0;
+  SPHERICAL = 1;
+}
+
+/**
+ * A custom WKB-encoded geometry data to be used in geometry statistics.
+ * The geometry may be a polygon to encode an s2 or h3 covering to provide
+ * vendor-agnostic coverings, or an evelope of geometries when a bounding
+ * box cannot be built (e.g. a geometry has spherical edges, or if an edge
+ * of geographic coordinates crosses the antimeridian).
+ */
+struct Geometry {
+  /** Bytes of a WKB-encoded geometry */
+  1: required binary geometry;
+  /**
+   * Edges of the geometry if it is a polygon. It may be different to the
+   * edges attribute from the GEOMETRY logical type.
+   */
+  2: optional Edges edges;
+}
+
+/**
+ * Bounding box of geometries in the representation of min/max value pair of
+ * coordinates from each axis. Values of Z and M are omitted for 2D geometries.
+ */
+struct BoundingBox {
+  1: required double xmin;
+  2: required double xmax;
+  3: required double ymin;
+  4: required double ymax;
+  5: optional double zmin;
+  6: optional double zmax;
+  7: optional double mmin;
+  8: optional double mmax;
+}
+
+struct Covering {
+  optional BoundingBox bbox    // A bounding box of geometries if it can be built.
+  optional Geometry covering   // A covering polygon of geometries if bbox is unavailable.
+}
+
+/** Statistics specific to GEOMETRY logical type */
+struct GeometryStatistics {
+  /** Covering of geometries */
+  1: optional Covering covering;
+
+  /**
+   * The geometry types of all geometries, or an empty array if they are not
```

Comment:
empty array or geometry types are not populated or both indicate unknown?

Reply:
I copied it from the GeoParquet specs: `An empty array explicitly signals that the geometry types are not known.`

Reply:
`geometry_types` in the GeoParquet is a list of string. Should we use enum instead? However, we need to define items for different dimensions, e.g. "POINT" and "POINT Z".

Reply:
It sounds like we can directly define enum like below:
```
enum GeometrySubType {
  POINT = 1;
  LINESTRING = 2;
  POLYGON = 3;
  MULTIPOINT = 4;
  MULTILINESTRING = 5;
  MULTIPOLYGON = 6;
  GEOMETRYCOLLECTION = 7;

  POINT_Z = 1001;
  LINESTRING_Z = 1002;
  POLYGON_Z = 1003;
  MULTIPOINT_Z = 1004;
  MULTILINESTRING_Z = 1005;
  MULTIPOLYGON_Z = 1006;
  GEOMETRYCOLLECTION_Z = 1007;

  POINT_M = 2001;
  LINESTRING_M = 2002;
  POLYGON_M = 2003;
  MULTIPOINT_M = 2004;
  MULTILINESTRING_M = 2005;
  MULTIPOLYGON_M = 2006;
  GEOMETRYCOLLECTION_M = 2007;

  POINT_ZM = 3001;
  LINESTRING_ZM = 3002;
  POLYGON_ZM = 3003;
  MULTIPOINT_ZM = 3004;
  MULTILINESTRING_ZM = 3005;
  MULTIPOLYGON_ZM = 3006;
  GEOMETRYCOLLECTION_ZM = 3007;
}
```

If the above list if too verbose and hard to maintain. We can simply use `list<i32>` to hold integer codes.